### PR TITLE
11790_home_page 2

### PR DESCRIPTION
**Pull request by tuananhp-1844**
https://edu-redmine.sun-asterisk.vn/issues/11790

### CODE REVIEW

File: app/Http/Controllers/Questions/QuestionController.php [hunk 1]
```diff
@@ -4,17 +4,44 @@
...
+use App\Repositories\Contracts\QuestionRepositoryInterface;
+use App\Repositories\Contracts\CategoryRepositoryInterface;
+use App\Http\Requests\Questions\CreateQuestionRequest;
 
 class QuestionController extends Controller
 {
+    private $questionRepository;
+    private $categoryRepository;
+
+    public function __construct(QuestionRepositoryInterface $questionRepository, CategoryRepositoryInterface $categoryRepository) {
```
Comment (framgiaciv3) [inline app/Http/Controllers/Questions/QuestionController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.Files.LineLength.TooLong</br>**Severity**: warning</br>**File**: app/Http/Controllers/Questions/QuestionController.php</br>**Line**: 16</br>Line exceeds 120 characters; contains 131 characters

Comment (framgiaciv3) [inline app/Http/Controllers/Questions/QuestionController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Functions.MultiLineFunctionDeclaration.BraceOnSameLine</br>**Severity**: error</br>**File**: app/Http/Controllers/Questions/QuestionController.php</br>**Line**: 16</br>Opening brace should be on a new line

File: app/Http/Requests/Languages/LanguageRequest.php [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        return true;
+    }
+
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array
+     */
+    public function rules()
+    {
```
Comment (framgiaciv3) [inline app/Http/Requests/Languages/LanguageRequest.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoComma</br>**Severity**: error</br>**File**: app/Http/Requests/Languages/LanguageRequest.php</br>**Line**: 27</br>Each line in an array declaration must end in a comma

File: app/Http/Requests/Questions/CreateQuestionRequest.php [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        return Auth::check();
+    }
+
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array
+     */
+    public function rules()
+    {
```
Comment (framgiaciv3) [inline app/Http/Requests/Questions/CreateQuestionRequest.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Http/Requests/Questions/CreateQuestionRequest.php</br>**Line**: 28</br>String "category" does not require double quotes; use single quotes instead

Comment (framgiaciv3) [inline app/Http/Requests/Questions/CreateQuestionRequest.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Http/Requests/Questions/CreateQuestionRequest.php</br>**Line**: 28</br>String "required" does not require double quotes; use single quotes instead

File: app/Http/Requests/Questions/CreateQuestionRequest.php [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        return [
```
Comment (framgiaciv3) [inline app/Http/Requests/Questions/CreateQuestionRequest.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Http/Requests/Questions/CreateQuestionRequest.php</br>**Line**: 29</br>String "title" does not require double quotes; use single quotes instead

Comment (framgiaciv3) [inline app/Http/Requests/Questions/CreateQuestionRequest.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Http/Requests/Questions/CreateQuestionRequest.php</br>**Line**: 29</br>String "required" does not require double quotes; use single quotes instead

File: app/Http/Requests/Questions/CreateQuestionRequest.php [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+            "category" => "required",
```
Comment (framgiaciv3) [inline app/Http/Requests/Questions/CreateQuestionRequest.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Http/Requests/Questions/CreateQuestionRequest.php</br>**Line**: 30</br>String "content" does not require double quotes; use single quotes instead

Comment (framgiaciv3) [inline app/Http/Requests/Questions/CreateQuestionRequest.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Http/Requests/Questions/CreateQuestionRequest.php</br>**Line**: 30</br>String "required" does not require double quotes; use single quotes instead

File: app/Http/Requests/Questions/CreateQuestionRequest.php [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+            "title" => "required",
```
Comment (framgiaciv3) [inline app/Http/Requests/Questions/CreateQuestionRequest.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Http/Requests/Questions/CreateQuestionRequest.php</br>**Line**: 31</br>String "ask" does not require double quotes; use single quotes instead

Comment (framgiaciv3) [inline app/Http/Requests/Questions/CreateQuestionRequest.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoComma</br>**Severity**: error</br>**File**: app/Http/Requests/Questions/CreateQuestionRequest.php</br>**Line**: 31</br>Each line in an array declaration must end in a comma

Comment (framgiaciv3) [inline app/Http/Requests/Questions/CreateQuestionRequest.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Http/Requests/Questions/CreateQuestionRequest.php</br>**Line**: 31</br>String "array" does not require double quotes; use single quotes instead

File: app/Providers/AppServiceProvider.php [hunk 7]
```diff
@@ -14,7 +14,16 @@ class AppServiceProvider extends ServiceProvider
      */
     public function register()
     {
-        //
+        $models = [
+            'Question',
+            'Category'
```
Comment (framgiaciv3) [inline app/Providers/AppServiceProvider.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoCommaAfterLast</br>**Severity**: error</br>**File**: app/Providers/AppServiceProvider.php</br>**Line**: 19</br>Comma required after last value in array declaration

File: app/Providers/AppServiceProvider.php [hunk 8]
```diff
@@ -14,7 +14,16 @@ class AppServiceProvider extends ServiceProvider
...
+        ];
+        foreach ($models as $key => $model) {
+            $this->app->bind(
+				"App\\Repositories\\Contracts\\{$model}RepositoryInterface",
```
Comment (framgiaciv3) [inline app/Providers/AppServiceProvider.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.WhiteSpace.DisallowTabIndent.TabsUsed</br>**Severity**: error</br>**File**: app/Providers/AppServiceProvider.php</br>**Line**: 23</br>Spaces must be used to indent lines; tabs are not allowed

File: app/Providers/AppServiceProvider.php [hunk 9]
```diff
@@ -14,7 +14,16 @@ class AppServiceProvider extends ServiceProvider
...
+				"App\\Repositories\\Eloquents\\{$model}Repository"
```
Comment (framgiaciv3) [inline app/Providers/AppServiceProvider.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.WhiteSpace.DisallowTabIndent.TabsUsed</br>**Severity**: error</br>**File**: app/Providers/AppServiceProvider.php</br>**Line**: 24</br>Spaces must be used to indent lines; tabs are not allowed

File: app/Providers/AppServiceProvider.php [hunk 10]
```diff
@@ -14,7 +14,16 @@ class AppServiceProvider extends ServiceProvider
...
+			);
```
Comment (framgiaciv3) [inline app/Providers/AppServiceProvider.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.WhiteSpace.DisallowTabIndent.TabsUsed</br>**Severity**: error</br>**File**: app/Providers/AppServiceProvider.php</br>**Line**: 25</br>Spaces must be used to indent lines; tabs are not allowed

File: app/Providers/AppServiceProvider.php [hunk 11]
```diff
@@ -14,7 +14,16 @@ class AppServiceProvider extends ServiceProvider
...
     public function register()
     {
-        //
+        $models = [
+            'Question',
+            'Category',
+        ];
+        foreach ($models as $key => $model) {
+            $this->app->bind(
+				'App\\Repositories\\Contracts\\{$model}RepositoryInterface',
```
Comment (framgiaciv3) [inline app/Providers/AppServiceProvider.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.WhiteSpace.DisallowTabIndent.TabsUsed</br>**Severity**: error</br>**File**: app/Providers/AppServiceProvider.php</br>**Line**: 23</br>Spaces must be used to indent lines; tabs are not allowed

File: app/Providers/AppServiceProvider.php [hunk 12]
```diff
@@ -14,7 +14,16 @@ class AppServiceProvider extends ServiceProvider
...
+				'App\\Repositories\\Eloquents\\{$model}Repository'
```
Comment (framgiaciv3) [inline app/Providers/AppServiceProvider.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.WhiteSpace.DisallowTabIndent.TabsUsed</br>**Severity**: error</br>**File**: app/Providers/AppServiceProvider.php</br>**Line**: 24</br>Spaces must be used to indent lines; tabs are not allowed

File: app/Providers/AppServiceProvider.php [hunk 13]
```diff
@@ -14,7 +14,16 @@ class AppServiceProvider extends ServiceProvider
...
+			);
```
Comment (framgiaciv3) [inline app/Providers/AppServiceProvider.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.WhiteSpace.DisallowTabIndent.TabsUsed</br>**Severity**: error</br>**File**: app/Providers/AppServiceProvider.php</br>**Line**: 25</br>Spaces must be used to indent lines; tabs are not allowed

File: app/Repositories/Contracts/CategoryRepositoryInterface.php [hunk 14]
```diff
@@ -0,0 +1,6 @@
+<?php
+namespace App\Repositories\Contracts;
+
+interface CategoryRepositoryInterface
```
Comment (framgiaciv3) [inline app/Repositories/Contracts/CategoryRepositoryInterface.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Files.EndFileNewline.NoneFound</br>**Severity**: error</br>**File**: app/Repositories/Contracts/CategoryRepositoryInterface.php</br>**Line**: 6</br>Expected 1 newline at end of file; 0 found

File: app/Repositories/Contracts/QuestionRepositoryInterface.php [hunk 15]
```diff
@@ -0,0 +1,10 @@
+<?php
+namespace App\Repositories\Contracts;
+
+interface QuestionRepositoryInterface
+{
+    public function newest();
+    public function unResolve();
+    public function isPoll();
```
Comment (framgiaciv3) [inline app/Repositories/Contracts/QuestionRepositoryInterface.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Files.EndFileNewline.NoneFound</br>**Severity**: error</br>**File**: app/Repositories/Contracts/QuestionRepositoryInterface.php</br>**Line**: 10</br>Expected 1 newline at end of file; 0 found

File: app/Repositories/Eloquents/BaseRepository.php [hunk 16]
```diff
@@ -0,0 +1,89 @@
...
+use Illuminate\Container\Container as App;
+use Illuminate\Database\Eloquent\Model;
+
+abstract class BaseRepository implements BaseRepositoryInterface
+{
+    /** var
+     */
+    protected $model;
+
+    public function __construct()
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/BaseRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.WhiteSpace.DisallowTabIndent.TabsUsed</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/BaseRepository.php</br>**Line**: 17</br>Spaces must be used to indent lines; tabs are not allowed

File: app/Repositories/Eloquents/BaseRepository.php [hunk 17]
```diff
@@ -0,0 +1,89 @@
...
+    {
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/BaseRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.WhiteSpace.DisallowTabIndent.TabsUsed</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/BaseRepository.php</br>**Line**: 18</br>Spaces must be used to indent lines; tabs are not allowed

File: app/Repositories/Eloquents/BaseRepository.php [hunk 18]
```diff
@@ -0,0 +1,89 @@
...
+		$this->setModel();
+	}
+
+    abstract public function getModel();
+
+    public function setModel()
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/BaseRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.WhiteSpace.DisallowTabIndent.TabsUsed</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/BaseRepository.php</br>**Line**: 24</br>Spaces must be used to indent lines; tabs are not allowed

File: app/Repositories/Eloquents/BaseRepository.php [hunk 19]
```diff
@@ -0,0 +1,89 @@
...
+    {
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/BaseRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.WhiteSpace.DisallowTabIndent.TabsUsed</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/BaseRepository.php</br>**Line**: 25</br>Spaces must be used to indent lines; tabs are not allowed

File: app/Repositories/Eloquents/BaseRepository.php [hunk 20]
```diff
@@ -0,0 +1,89 @@
...
+		$this->model = app()->make(
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/BaseRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.WhiteSpace.DisallowTabIndent.TabsUsed</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/BaseRepository.php</br>**Line**: 26</br>Spaces must be used to indent lines; tabs are not allowed

File: app/Repositories/Eloquents/BaseRepository.php [hunk 21]
```diff
@@ -0,0 +1,89 @@
...
+			$this->getModel()
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/BaseRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.WhiteSpace.DisallowTabIndent.TabsUsed</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/BaseRepository.php</br>**Line**: 27</br>Spaces must be used to indent lines; tabs are not allowed

File: app/Repositories/Eloquents/CategoryRepository.php [hunk 22]
```diff
@@ -0,0 +1,17 @@
...
+
+namespace App\Repositories\Eloquents;
+
+use App\Repositories\Contracts\CategoryRepositoryInterface;
+use App\Models\Category;
+
+class CategoryRepository extends BaseRepository implements CategoryRepositoryInterface
+{
+    /**
+     * Specify Model class name
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/CategoryRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Scope.MethodScope.Missing</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/CategoryRepository.php</br>**Line**: 13</br>Visibility must be declared on method "getModel"

File: app/Repositories/Eloquents/QuestionRepository.php [hunk 23]
```diff
@@ -0,0 +1,51 @@
...
+
+use App\Repositories\Contracts\QuestionRepositoryInterface;
+use App\Models\Question;
+use Illuminate\Http\Request;
+use Auth;
+
+class QuestionRepository extends BaseRepository implements QuestionRepositoryInterface
+{
+    /**
+     * Specify Model class name
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/QuestionRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Scope.MethodScope.Missing</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/QuestionRepository.php</br>**Line**: 15</br>Visibility must be declared on method "getModel"

File: app/Repositories/Eloquents/CategoryRepository.php (hunk 22 above)
Comment (framgiaciv3) [inline app/Repositories/Eloquents/CategoryRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Scope.MethodScope.Missing</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/CategoryRepository.php</br>**Line**: 13</br>Visibility must be declared on method "getModel"

File: app/Providers/AppServiceProvider.php [hunk 24]
```diff
@@ -14,7 +14,15 @@ class AppServiceProvider extends ServiceProvider
...
     public function register()
     {
-        //
+        $models = [
+            'Question',
+            'Category',
+        ];
+        foreach ($models as $key => $model) {
+            $this->app->bind(
+            'App\\Repositories\\Contracts\\{$model}RepositoryInterface',
```
Comment (framgiaciv3) [inline app/Providers/AppServiceProvider.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.Indent</br>**Severity**: error</br>**File**: app/Providers/AppServiceProvider.php</br>**Line**: 23</br>Multi-line function call not indented correctly; expected 16 spaces but found 12

File: app/Providers/AppServiceProvider.php [hunk 25]
```diff
@@ -14,7 +14,15 @@ class AppServiceProvider extends ServiceProvider
...
+            'App\\Repositories\\Eloquents\\{$model}Repository');
```
Comment (framgiaciv3) [inline app/Providers/AppServiceProvider.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.CloseBracketLine</br>**Severity**: error</br>**File**: app/Providers/AppServiceProvider.php</br>**Line**: 24</br>Closing parenthesis of a multi-line function call must be on a line by itself

**Comment by tuananhp-1844** [comment]
ready

File: app/Http/Controllers/Languages/LanguageController.php [hunk 26]
```diff
@@ -0,0 +1,22 @@
...
+    /**
+     * Handle the incoming request.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function __invoke(LanguageRequest $request)
+    {
+        session(['website_language' => $request->lang]);
+        return redirect()->back();
```
Comment (BunnyPi04) [inline app/Http/Controllers/Languages/LanguageController.php]: thiếu dòng trắng trước return

File: public/css/custom.css [hunk 27]
```diff
@@ -0,0 +1,9 @@
+.error{
+    color:red !important
+}
+.editor-toolbar {
+    margin-left: 190px
+}
+.CodeMirror {
+    margin-left: 190px
+}
```
Comment (BunnyPi04) [inline public/css/custom.css]: thiếu dòng trắng cuối file
thiếu chấm phẩy cuối dòng
tên class viết thường thôi

File: resources/views/layouts/footer.blade.php [hunk 28]
```diff
@@ -0,0 +1,78 @@
+<footer id="footer">
+		<section class="container">
```
Comment (BunnyPi04) [inline resources/views/layouts/footer.blade.php]: indent = 4 spaces

**Comment by tuananhp-1844** [comment]
ready

File: app/Http/Controllers/Questions/QuestionController.php [hunk 29]
```diff
@@ -4,17 +4,46 @@
...
+                $questions = $questions->isPoll();
+                break;
+            case 'no-answer':
+                # code...
+                break;
+            default:
+                $questions = $questions->newest();
+                break;
+        }
+        $questions = $questions->paginate(10);
```
Comment (BunnyPi04) [inline app/Http/Controllers/Questions/QuestionController.php]: paginate đưa vào config nhé

File: app/Http/Requests/Languages/LanguageRequest.php [hunk 30]
```diff
@@ -0,0 +1,37 @@
...
+        return [
+            'lang' => 'in:vi,en',
+        ];
+    }
+
+    public function all($keys = null)
+    {
+        $data = parent::all($keys);
+        $data['lang'] = $this->route('lang');
+        return $data;
```
Comment (BunnyPi04) [inline app/Http/Requests/Languages/LanguageRequest.php]: đây vẫn thiếu dòng trắng trước return

File: resources/views/layouts/footer.blade.php [hunk 31]
```diff
@@ -0,0 +1,86 @@
+<footer id="footer">
+	<section class="container">
```
Comment (BunnyPi04) [inline resources/views/layouts/footer.blade.php]: indent = 4 spaces chưa được

File: resources/views/layouts/footer.blade.php [hunk 32]
```diff
@@ -0,0 +1,86 @@
...
+		<div class="row">
+			<div class="col-md-4">
+				<div class="widget widget_contact">
+					<h3 class="widget_title">Where We Are ?</h3>
+					<p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Morbi adipiscing gravida odio, sit amet
+						suscipit risus ultrices eu.</p>
+					<ul>
+						<li>
+							<span>Address :</span>
```
Comment (BunnyPi04) [inline resources/views/layouts/footer.blade.php]: i18n

**Comment by tuananhp-1844** [comment]
ready

File: resources/views/home.blade.php [hunk 33]
```diff
@@ -5,16 +5,15 @@
     <div class="row justify-content-center">
         <div class="col-md-8">
             <div class="card">
-                <div class="card-header">Dashboard</div>
+                <div class="card-header">{{__('Dashboard')}}</div>
```
Comment (BunnyPi04) [inline resources/views/home.blade.php]: convention

File: resources/views/layouts/header.blade.php [hunk 34]
```diff
@@ -0,0 +1,124 @@
+<div class="panel-pop" id="signup">
+    <form action="/file-upload" class="dropzone" id="my-awesome-dropzone"></form>
```
Comment (BunnyPi04) [inline resources/views/layouts/header.blade.php]: sử dụng đường dẫn tuyệt đối của laravel nhé

File: resources/views/home.blade.php (hunk 33 above)
Comment (loind-1875) [inline resources/views/home.blade.php]: convention

File: resources/views/home.blade.php [hunk 35]
```diff
@@ -5,16 +5,15 @@
...
-                        <div class="alert alert-success" role="alert">
-                            {{ session('status') }}
-                        </div>
+                    <div class="alert alert-success" role="alert">
+                        {{ session('status') }}
+                    </div>
                     @endif
-
-                    You are logged in!
+                    {{__('You are logged in!')}}
```
Comment (loind-1875) [inline resources/views/home.blade.php]: convention

File: resources/views/layouts/app.blade.php [hunk 36]
```diff
@@ -1,75 +1,79 @@
...
-	<!-- Responsive Style -->
-	<link rel="stylesheet" href="{{ asset('bower_components/asset-project1-sun/css/responsive.css') }}">
-	
-	<!-- Favicons -->
-	<link rel="shortcut icon" href="{{ asset('bower_components/asset-project1-sun/images/favicon.png') }}">
-  
-	<link rel="stylesheet" href="{{ asset('css/custom.css') }}">
+    <!-- Basic Page Needs -->
+    <meta charset="utf-8">
+    <title>{{__('Ask me – Responsive Questions and Answers Template')}}</title>
```
Comment (loind-1875) [inline resources/views/layouts/app.blade.php]: tương tự các chỗ ntn

File: resources/views/layouts/footer.blade.php [hunk 37]
```diff
@@ -0,0 +1,88 @@
+<footer id="footer">
+    <section class="container">
+        <div class="row">
+            <div class="col-md-4">
+                <div class="widget widget_contact">
+                    <h3 class="widget_title">{{__('Where We Are ?')}}</h3>
+                    <p>{{__('Lorem ipsum dolor sit amet, consectetur adipiscing elit. Morbi adipiscing gravida odio, sit amet
+						suscipit risus ultrices eu.')}}</p>
```
Comment (loind-1875) [inline resources/views/layouts/footer.blade.php]: xuồn dòng linh tinh thế này

File: resources/views/layouts/footer.blade.php [hunk 38]
```diff
@@ -0,0 +1,88 @@
...
+        </div><!-- End row -->
+    </section><!-- End container -->
+</footer><!-- End footer -->
+<footer id="footer-bottom">
+    <section class="container">
+        <div class="copyrights f_left">{{__('Copyright 2014 Ask me')}} | <a href="#">{{__('By 2code')}}</a></div>
+        <div class="social_icons f_right">
+            <ul>
+                <li class="twitter"><a original-title="Twitter" class="tooltip-n" href="#"><i
+                            class="social_icon-twitter font17"></i></a></li>
```
Comment (loind-1875) [inline resources/views/layouts/footer.blade.php]: tương tự

File: resources/views/layouts/header.blade.php [hunk 39]
```diff
@@ -0,0 +1,124 @@
...
+    <h2>{{__('Register Now')}}<i class="icon-remove"></i></h2>
+    <div class="form-style form-style-3">
+        <form>
+            <div class="form-inputs clearfix">
+                <p>
+                    <label class="required">{{__('Username')}}<span>*</span></label>
```
Comment (loind-1875) [inline resources/views/layouts/header.blade.php]: k i18n à

**Comment by tuananhp-1844** [comment]
ready

File: app/Http/Controllers/Questions/QuestionController.php [hunk 40]
```diff
@@ -4,17 +4,46 @@
...
+                $questions = $questions->isPoll();
+                break;
+            case 'no-answer':
+                # code...
+                break;
+            default:
+                $questions = $questions->newest();
+                break;
+        }
+        $questions = $questions->paginate(config('pagination.question', 10));
```
Comment (BunnyPi04) [inline app/Http/Controllers/Questions/QuestionController.php]: đưa config kiểu gì thế?

File: resources/views/layouts/header.blade.php [hunk 41]
```diff
@@ -0,0 +1,126 @@
...
+                <button type="submit" class="search-submit"></button>
+            </form>
+        </div>
+    </section><!-- End container -->
+</div><!-- End header-top -->
+<header id="header">
+    <section class="container clearfix">
+        <div class="logo">
+            <a href="{{ route('home')}}">
+                <img alt="" src="{{ asset('bower_components/asset-project1-sun/images/logo.png') }}">
```
Comment (BunnyPi04) [inline resources/views/layouts/header.blade.php]: base link đưa vào config

**Pull request merged**